module_utils/network/junos/config/l3_interfaces/l3_interfaces.py

```diff
@@ -0,0 +1,244 @@
+# Copyright (C) 2019  Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""
+The junos_l3_interfaces class
+It is in this file where the current configuration (as dict)
+is compared to the provided configuration (as dict) and the command set
+necessary to bring the current configuration to it's desired end-state is
+created
+"""
+
+from ansible.module_utils.network.common.utils import to_list
+
+from ansible.module_utils.network. \
+    junos.argspec.l3_interfaces.l3_interfaces import L3_interfacesArgs
+from ansible.module_utils.network.junos.config.base import ConfigBase
+from ansible.module_utils.network.junos.junos import (locked_config,
+                                                      load_config,
+                                                      commit_configuration,
+                                                      discard_changes,
+                                                      tostring)
+from ansible.module_utils.network.junos.facts.facts import Facts
+from ansible.module_utils.network.junos.utils import utils as junosutils
+
+
+class L3_interfaces(ConfigBase, L3_interfacesArgs):
+    """
+    The junos_l3_interfaces class
+    """
+
+    gather_subset = [
+        '!all',
+        '!min',
+    ]
+
+    gather_network_resources = [
+        'l3_interfaces',
+    ]
+
+    def get_l3_interfaces_facts(self):
+        """ Get the 'facts' (the current configuration)
+
+        :rtype: A dictionary
+        :returns: The current configuration as a dictionary
+        """
+        facts, _warnings = Facts().get_facts(self._module,
```

Comment:
This needs to be refactored as per the updated changes in ansible/ansible and RMB builder code